### DISCUSSION TITLE
Colima is not using vmType VZ

### DISCUSSION THREAD

**Discussion by anazar2001** [abiosoft/colima#576] [Q&A]
Updated my mac to MacOs 13.1 and colima is still using qemu even with brand new profile. What am I missing?

colima start vz --arch x86_64 --cpu 6 --memory 16 --disk 128 --vm-type=vz
INFO[0000] starting colima [profile=vz]
INFO[0000] runtime: docker
INFO[0000] creating and starting ...                     context=vm
> Terminal is not available, proceeding without opening an editor
> "Attempting to download the image from \"https://github.com/abiosoft/alpine-lima/releases/download/colima-v0.5.0-2/alpine-lima-clm-3.16.2-x86_64.iso\"" digest="sha512:e9e118498f3a0745574ffc3686105d2c9777f7142164fe50ee47909dabd504c80ac29aeb76f9582706173310d1488d3b6f0ee9018e2a6aadc28eefb7767b63ec"
> Using cache "/Users/andrei.nazarenka/Library/Caches/lima/download/by-url-sha256/e1cbaef56fc8e75af106d934389a17618643fcecfa066302d45ffb0390709851/data"
> [hostagent] Starting QEMU (hint: to watch the boot progress, see "/Users/andrei.nazarenka/.lima/colima-vz/serial.log")
> SSH Local Port: 51361

**Comment by abiosoft** [comment]
Virtualisation.framework is not yet as stable as QEMU, it was removed as the default in v0.5.1.

You need to specify `--vm-type vz` to make use of it.

**Reply by anazar2001** [reply]
I am using that Param as you can see from my original post

**Reply by abiosoft** [reply]
@anazar2001 sorry, missed that. 

It is not supported for cross architecture. i.e. x86_64 on m1.

**Reply by anazar2001** [reply]
Ok. Is there any other way to speedup cross architecture setup? Have to run Oracle on M1. It works but is quite slow

**Reply by abiosoft** [reply]
It is actually a limitation from Apple. The API for virtual machines does not support cross architecture yet.

**Reply by anazar2001** [reply]
I understand. But is there any way or workaround to make cross platform scenario work faster?

**Reply by Sangeppato** [reply]
@anazar2001 How about using Rosetta in an VZ VM? You can enable it by passing `--vz-rosetta`

**Reply by anazar2001** [reply]
Rosetta only helps to run x86 images on arm platform. I need to run x86 image on x86 on M1 Mac. Oracle image is only available for x86

**Reply by Sangeppato** [reply]
I'm sorry, maybe I'm missing something.. but isn't the underlying arch irrelevant from the container POV?

**Reply by anazar2001** [reply]
Oracle has some low level arch limitations. If not Oracle it all would not be an issue

**Reply by Sangeppato** [reply]
I see..
In that case I don't think there's any way around using QEMU for emulation, and that is always going to be slow

**Reply by anazar2001** [reply]
I hoped that with vz it is going to be some faster and lightway emulation/virtualization process.

**Reply by Sangeppato** [reply]
"Virtualization" is only really possible if the host and guest archs match, otherwise you're back at emulation... And since Rosetta 2 doesn't support running a whole different-arch OS the only real option is QEMU.

The real issue here is that Oracle is doing something "weird", because theoretically it should run with Rosetta just fine..

Out of curiosity, what does docker tell you if you try to run the container on an VZ VM with Rosetta enabled?

**Reply by anazar2001** [reply]
To be honest I didn't try since I thought Rosetta is useful for x86->arm which is not my scenario. If I try oracle in arm arch it even says it is started but just doesn't work

**Reply by Sangeppato** [reply]
I'm sorry but to me it looks like x86->arm is exactly your scenario.
What do you mean with "it doesn't work"? What does the container log say?

**Reply by anazar2001** [reply]
I will try that again. But Oracle image will not work for arm. You can Google it. It is a known issue related to Oracle using some low level os feature which is not translated properly by rosetta
